src/components/actions/processTrigger.ts

```diff
@@ -0,0 +1,101 @@
+import { ActionInTemplate, TriggerPayload, ActionSequential } from '../../types'
+import DBConnect from '../databaseConnect'
+import { actionLibrary } from '../pluginsConnect'
+import { EvaluatorNode } from '@openmsupply/expression-evaluator/lib/types'
+import { executeAction } from './executeAction'
+import { ActionQueueStatus, TriggerQueueStatus } from '../../generated/graphql'
+import { swapOutAliasedActions } from './helpers'
+
+// Dev config
+const showActionOutcomeLog = false
+
+export async function processTrigger(payload: TriggerPayload) {
+  const { trigger_id, trigger, table, record_id, data, event_code } = payload
+
+  const templateId = await DBConnect.getTemplateIdFromTrigger(payload.table, payload.record_id)
+
+  // Get Actions from matching Template (and match templateActionCode if applicable)
+  const actions = (await DBConnect.getActionsByTemplateId(templateId, trigger))
+    .filter((action) => {
+      if (!event_code) return true
+      else return action.event_code === event_code
+    })
+    .map((action) => (action.code !== 'alias' ? action : swapOutAliasedActions(templateId, action)))
+
+  const resolvedActions = await Promise.all(actions)
+
+  // Separate into Sequential and Async actions
+  const actionsSequential: ActionSequential[] = []
+  const actionsAsync: ActionInTemplate[] = []
+  for (const action of resolvedActions) {
+    if (action.sequence) actionsSequential.push(action as ActionSequential)
+    else actionsAsync.push(action)
+  }
```

Comment:
Personal stylistic preference

```suggestion
// Separate into Sequential and Async actions
  const actionsSequential = resolvedActions.filter(
    ({ sequence }) => !!sequence
  ) as ActionSequential[]
  const actionsAsync = resolvedActions.filter(({ sequence }) => !sequence)
```

Reply:
Even cleaner now that we can remove the `as`: 5af9f91b